src/prebid.js

```diff
@@ -384,43 +351,30 @@ $$PREBID_GLOBAL$$.requestBids = function ({ bidsBackHandler, timeout, adUnits, a
     adUnit.bids = adUnit.bids.filter(nativeBidder);
   });
 
-  if (auctionRunning) {
-    bidRequestQueue.push(() => {
-      $$PREBID_GLOBAL$$.requestBids({ bidsBackHandler, timeout: cbTimeout, adUnits, adUnitCodes });
-    });
-    return;
-  }
-
-  auctionRunning = true;
-
-  // we will use adUnitCodes for filtering the current auction
-  $$PREBID_GLOBAL$$._adUnitCodes = adUnitCodes;
-
-  bidmanager.externalCallbackReset();
-  clearPlacements();
-
   if (!adUnits || adUnits.length === 0) {
     utils.logMessage('No adUnits configured. No bids requested.');
     if (typeof bidsBackHandler === 'function') {
-      bidmanager.addOneTimeCallback(bidsBackHandler, false);
+      // executeCallback, this will only be called in case of first request
+      try {
+        bidsBackHandler();
+      } catch (e) {
+        utils.logError('Error executing bidsBackHandler', null, e);
+      } finally {
+        $$PREBID_GLOBAL$$.clearAuction();
+      }
     }
-    bidmanager.executeCallback();
     return;
   }
 
-  // set timeout for all bids
-  const timedOut = true;
-  const timeoutCallback = bidmanager.executeCallback.bind(bidmanager, timedOut);
-  const timer = setTimeout(timeoutCallback, cbTimeout);
-  setAjaxTimeout(cbTimeout);
-  if (typeof bidsBackHandler === 'function') {
-    bidmanager.addOneTimeCallback(bidsBackHandler, timer);
-  }
+  const auction = auctionManager.createAuction();
+  auction.setAdUnits(adUnits);
+  auction.setAdUnitCodes(adUnitCodes);
 
-  adaptermanager.callBids({ adUnits, adUnitCodes, cbTimeout });
-  if ($$PREBID_GLOBAL$$._bidsRequested.length === 0) {
-    bidmanager.executeCallback();
+  if (typeof bidsBackHandler === 'function') {
+    auction.startAuctionTimer(bidsBackHandler, cbTimeout);
   }
+  auction.callBids(cbTimeout);
+  setAjaxTimeout(cbTimeout);
```

Comment:
I don't think we can do this if we're going to allow multiple auctions simultaneously.